zilliqa/src/exec.rs

```diff
@@ -391,11 +393,7 @@ impl State {
             .append_handler_register(inspector_handle_register)
             .modify_cfg_env(|c| {
                 c.chain_id = chain_id;
-                // We disable the balance check (which ensures the from account's balance is greater than the
-                // transaction's `gas_price * gas_limit`), because Scilla transactions are often submitted with
-                // overly high `gas_limit`s. This is probably because there is no gas estimation feature for Scilla
-                // transactions.
-                c.disable_balance_check = true;
+                c.disable_base_fee = skip_base_fee_check;
```

Comment:
What does ZQ1 do? Will check this afternoon if you don't get there first .. I'm a bit nervous about discounting the check altogether, though I do agree with the logic.

Reply:
For `eth_call`s, ZQ1 sets the `gas_price` to the DS block gas limit. That achieves the same thing as skipping the fee check.

ZQ1 doesn't use REVM, so doesn't have the balance check. If we employed the same approach, all `eth_call`s would fail unless the sender had a balance of `gas_limit * block_gas_price`.